### PR TITLE
feat(crons): Resolve monitor issues

### LINKED ISSUE #59069: Add auto resolve to incidents

### PR DESCRIPTION
**Pull request by rjo100**
Resolves issues with thresholds as appropriate

Builds off of https://github.com/getsentry/sentry/pull/58293

Closes https://github.com/getsentry/sentry/issues/59069

### CODE REVIEW

File: src/sentry/monitors/logic/mark_failed.py [hunk 1]
```diff
@@ -301,7 +300,7 @@ def create_issue_platform_occurrence(
         "contexts": {"monitor": get_monitor_environment_context(monitor_env)},
         "environment": monitor_env.environment.name,
         "event_id": occurrence.event_id,
-        "fingerprint": fingerprint
+        "fingerprint": [fingerprint]
```
Comment (evanpurkhiser) [inline src/sentry/monitors/logic/mark_failed.py:303]: Why did this have to change?

File: src/sentry/monitors/logic/mark_ok.py [hunk 2]
```diff
@@ -43,22 +43,49 @@ def mark_ok(checkin: MonitorCheckIn, ts: datetime):
                 previous_checkin["status"] == CheckInStatus.OK
                 for previous_checkin in previous_checkins
             )
+        else:
+            incident_recovering = True
```
Comment (evanpurkhiser) [inline src/sentry/monitors/logic/mark_ok.py:47]: Does this mean we're just __always__ resolving issues for monitors that don't use thresholds when an OK check-in comes in now?

File: src/sentry/monitors/logic/mark_failed.py (hunk 1 above)
Comment (rjo100) [reply src/sentry/monitors/logic/mark_failed.py:303]: separate fix. can break it out. fingerprint was invalid only noticed it during testing

File: src/sentry/monitors/logic/mark_ok.py (hunk 2 above)
Comment (rjo100) [reply src/sentry/monitors/logic/mark_ok.py:47]: well we just run an update which can be a no-op

Comment (rjo100) [reply src/sentry/monitors/logic/mark_ok.py:47]: also there could be a failure threshold but not a recovery one and we need to cover that

Comment (rjo100) [reply src/sentry/monitors/logic/mark_ok.py:47]: i guess previously we weren't closing all incidents correctly if there's a failure threshold but not a recovery one

Comment (evanpurkhiser) [reply src/sentry/monitors/logic/mark_ok.py:47]: I just mean, the previous behavior was that monitors that do not have thresholds never have their issues resolved right. How it looks like we're actively resolving issues for those monitors no?

Comment (evanpurkhiser) [reply src/sentry/monitors/logic/mark_ok.py:47]: Let's go ahead and add a comment as to why we're setting this to true

File: src/sentry/monitors/logic/mark_ok.py [hunk 3]
```diff
@@ -43,22 +43,51 @@ def mark_ok(checkin: MonitorCheckIn, ts: datetime):
...
-                MonitorIncident.objects.filter(
-                    monitor_environment=monitor_env,
-                    grouphash=monitor_env.incident_grouphash,
-                ).update(
-                    resolving_checkin=checkin,
-                    resolving_timestamp=checkin.date_added,
-                )
+        # Resolve any open incidents
+        if incident_recovering:
+            # TODO(rjo100): Check for multiple open incidents due to a logic bug
```
Comment (rjo100) [inline src/sentry/monitors/logic/mark_ok.py:52]: needed to change this logic as we were only previously resolving incidents if `recovery_threshold` was set irrespective of if `failure_issue_threshold` was set. once we drain all orphaned incidents we can change this

Comment (evanpurkhiser) [reply src/sentry/monitors/logic/mark_ok.py:52]: Can you actually update this comment to what you wrote here in this comment haha

**Review by evanpurkhiser — APPROVED** [review]

**Pull request merged**